test/integration/targets/postgresql/tasks/main.yml

```diff
@@ -1,847 +1,82 @@
-    state: 'present'
-    conn_limit: '100'
-    encoding: 'LATIN1'
-    lc_collate: 'pt_BR{{ locale_latin_suffix }}'
-    lc_ctype: 'es_ES{{ locale_latin_suffix }}'
-    template: 'template0'
-    login_user: "{{ pg_user }}"
-
-- name: Check that the DB has all of our options
-  become_user: "{{ pg_user }}"
-  become: True
-  shell: echo "select datname, datconnlimit, pg_encoding_to_char(encoding), datcollate, datctype from pg_database where datname = '{{ db_name }}';" | psql -d postgres
-  register: result
-
-- assert:
-    that:
-      - "result.stdout_lines[-1] == '(1 row)'"
-      - "'LATIN1' in result.stdout_lines[-2]"
-      - "'pt_BR' in result.stdout_lines[-2]"
-      - "'es_ES' in result.stdout_lines[-2]"
-      - "'UTF8' not in result.stdout_lines[-2]"
-      - "'en_US' not in result.stdout_lines[-2]"
-      - "'100' in result.stdout_lines[-2]"
-
-- name: Check that running db cration with options a second time does nothing
-  become_user: "{{ pg_user }}"
-  become: True
-  postgresql_db:
-    name: '{{ db_name }}'
-    state: 'present'
-    conn_limit: '100'
-    encoding: 'LATIN1'
-    lc_collate: 'pt_BR{{ locale_latin_suffix }}'
-    lc_ctype: 'es_ES{{ locale_latin_suffix }}'
-    template: 'template0'
-    login_user: "{{ pg_user }}"
-  register: result
-
-- assert:
-    that:
-      - 'result.changed == False'
+# Test ssl.
+# Restricted using Debian family because of there are errors on other distributions
+# that not related with PostgreSQL or psycopg2 SSL support.
+# The tests' key point is to be sure that ssl options work in general
+- include_tasks: ssl.yml
+  when: ansible_os_family == 'Debian' and postgres_version_resp.stdout is version('9.4', '>=')
 
+- include_tasks: '{{ loop_item }}'
+  loop:
+  # Test postgresql_set
+  - postgresql_set.yml
 
-- name: Check that attempting to change encoding returns an error
-  become_user: "{{ pg_user }}"
-  become: True
-  postgresql_db:
-    name: '{{ db_name }}'
-    state: 'present'
-    encoding: 'UTF8'
-    lc_collate: 'pt_BR{{ locale_utf8_suffix }}'
-    lc_ctype: 'es_ES{{ locale_utf8_suffix }}'
-    template: 'template0'
-    login_user: "{{ pg_user }}"
-  register: result
-  ignore_errors: True
+  # Test postgresql_copy module
+  - postgresql_copy.yml
 
-- assert:
-    that:
-      - 'result.failed == True'
+  # Test postgresql_slot module.
+  # Physical replication slots are available from PostgreSQL 9.4
+  - postgresql_slot.yml
+  loop_control:
+    loop_var: loop_item
+  when: postgres_version_resp.stdout is version('9.4', '>=')
 
-- name: Check that changing the conn_limit actually works
-  become_user: "{{ pg_user }}"
-  become: True
-  postgresql_db:
-    name: '{{ db_name }}'
-    state: 'present'
-    conn_limit: '200'
-    encoding: 'LATIN1'
-    lc_collate: 'pt_BR{{ locale_latin_suffix }}'
-    lc_ctype: 'es_ES{{ locale_latin_suffix }}'
-    template: 'template0'
-    login_user: "{{ pg_user }}"
-  register: result
+- include_tasks: '{{ loop_item }}'
```

Comment:
You can use `include_tasks` with a loop here, but it would be better to list our several `import_task` steps since all the files are static and there is no conditional.